<launch>
  <arg name="raw_input" default="" />
  <arg name="rosbag_output" default="" />
  <arg name="use_relative_timestamp" default="false" />

  <node pkg="prophesee_ros_driver" type="prophesee_ros_publisher" name="prophesee_ros_publisher" respawn="false" output="screen">
    <param name="camera_name" value="camera" />
    <param name="publish_cd" value="true" />

    <!-- Path to the file with the camera settings -->
    <param name="bias_file" value=""/>

    <!-- Instead of live camera, you can read data offline from raw file -->
    <param name="raw_file_to_read" value="$(arg raw_input)" />

    <!-- Convert raw file into rosbag -->
    <param name="output_rosbag_path" value="$(arg rosbag_output)" />
    <param name="rosbag_event_topic" value="/prophesee/camera/cd_events_buffer" />

    <!-- Whether use relative timestamp for events. True: start with 0. False: start with current ROS time -->
    <param name="use_relative_timestamp" value="$(arg use_relative_timestamp)" />

    <!-- approximate time span in seconds for which events will
	 be aggregated until a ROS message is generated. Defaults to
	 100 usec -->
    <param name="event_delta_t" value="0.000100" />
  </node>

</launch>
